packages/server/src/utils.ts

```diff
@@ -0,0 +1,74 @@
+import sha256 from "crypto-js/sha256";
+import {AuthType, KeyType, Property, PropertyType, Signature} from "@0auth/message";
+import {ec as ECDSA, eddsa as EdDSA} from "elliptic";
+
+export function utf8ToBase64(str: string): string {
+  return Buffer.from(encodeURIComponent(str), "binary").toString("base64");
+}
+
+export function hashProperty(property: Property): string {
+  if (property.type == PropertyType.Hash)
+    return property.value;
+
+  return hash(utf8ToBase64(property.key)+":"+utf8ToBase64(property.value));
+}
+
+export function getMerkleRoot(properties: string[]): string {
```

Comment:
The merkle tree refers to a binary tree made of hash.
Merkle root is the root value of the tree.